src/lib.rs

```diff
@@ -202,8 +214,31 @@ impl Firebase {
     /// # }
     /// ```
     pub async fn set<T>(&self, data: &T) -> RequestResult<Response>
-        where
-            T: Serialize + DeserializeOwned + Debug,
+    where
+        T: Serialize + DeserializeOwned + Debug,
+    {
+        let data = serde_json::to_value(&data).unwrap();
+        self.request(Method::PUT, Some(data)).await
+    }
```

Comment:
This introduces a breaking change in the lib. I suggest leave `set` as the POST and create another `put` method that will do the PUT request.